rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,261 @@
+          //
+          // If someone wants CSS with modules, they can simply use the .sass file extension, since SASS
+          // is essentially a superset of CSS. (There's a small performance penalty for applying SASS to
+          // a plain CSS file, but the extra syntax validation probably justifies that cost.)
+          //
+          // The SASS docs are here: https://sass-lang.com/documentation/syntax
+          test: /\.(scss|sass|css)$/,
+          exclude: /node_modules/,
+          use: [
+            // "For production builds it's recommended to extract the CSS from your bundle being able to
+            // use parallel loading of CSS/JS resources later on. This can be achieved by using the
+            // mini-css-extract-plugin, because it creates separate css files."
+            //
+            // "For development mode (including webpack-dev-server) you can use style-loader, because it injects
+            // CSS into the DOM using multiple <style></style> and works faster."
+            //
+            // "WARNING: Do not use style-loader and mini-css-extract-plugin together."
+            production
+              ? {
+                  loader: MiniCssExtractPlugin.loader
+                }
+              : {
+                  // Generates JavaScript code that injects CSS styles into the DOM at runtime.
+                  // The default configuration creates <style> elements from JS strings
+                  // https://www.npmjs.com/package/style-loader
+                  loader: require.resolve('style-loader')
+                },
+
+            {
+              // Translates CSS into CommonJS
+              // https://www.npmjs.com/package/css-loader
+              loader: require.resolve('css-loader'),
+              options: {
+                // 0 => no loaders (default);
+                // 1 => postcss-loader;
+                // 2 => postcss-loader, sass-loader
+                importLoaders: 2,
+
+                // Enable CSS modules:  https://github.com/css-modules/css-modules
+                modules: {
+                  // The "auto" setting has a confusing design:
+                  // - "false" disables CSS modules, i.e. ":local" and ":global" selectors can't be used at all
+                  // - "true" means magically disable CSS modules if the file extension isn't like ".module.css"
+                  //   or ".module.scss"
+                  // - a lambda disables CSS modules only if the lambda returns false; the function parameter is
+                  //   the resource path
+                  // - a RegExp disables CSS modules only if the resource path does not match the RegExp
+                  //
+                  // NOTE: Counterintuitively, if you instead set "modules=true" then CSS modules are enabled
+                  //       without magic, equivalent to "auto: () => true" instead of "auto: true"
+                  //
+                  // DEFAULT: "true" (i.e. path based magic)
```

Comment:
This feels like it belongs in the css-modules documentation, not in our `webpack-base.config`. I suggest, instead:
```suggestion
                  // We define `auto` as a function so that we can selectively disable CSS modules only in the
                  // specific cases of (1) files with ".global." in the filename and (2) ".css" files.
                  // For all other files, we enable CSS modules. See the css-modules documentation for details.
```

Reply:
> This feels like it belongs in the css-modules documentation, not in our `webpack-base.config`.

We don't own in the `css-modules` project. I added these notes for posterity, to save someone else the headache of trying to understand the docs for that setting. :-)